types/index.d.ts

```diff
@@ -0,0 +1,160 @@
+import type { Client, Subscription } from 'aedes';
+import type { AedesPacket } from 'aedes-packet';
+import type { QoS } from 'mqtt-packet';
+import type { Readable } from 'stream';
+
+export type { AedesPacket as Packet } from 'aedes-packet';
+
+export interface AedesPersistenceSubscription {
+  clientId: string;
+  topic: string,
+  qos?: QoS
+}
+
+type LastHearthbeatTimestamp = Date;
+
+export interface Brokers {
+  [brokerId: string]: LastHearthbeatTimestamp;
+}
+
+export type CallbackError = Error | null | undefined;
+
+export type WillPacket = AedesPacket & { [key: string]: any };
+
+interface Incoming {
+  [clientId: string]: { [messageId: string]: AedesPacket };
+}
+
+export class AedesMemoryPersistence {
```

Comment:
You should add a `AedesPersistence` interface. The other packages will reuse the interface.

Reply:
Good point